apps/member-profile/app/shared/components/resource-form.tsx

```diff
@@ -136,20 +136,75 @@ export function ResourceTagsField({
   const createFetcher = useFetcher<unknown>();
   const listFetcher = useFetcher<SearchTagsResult>();
 
-  const [newTagId, setNewTagId] = useState<string>(id());
   const [search, setSearch] = useState<string>('');
+  const [selectedTags, setSelectedTags] = useState<Record<string, unknown>>({});
 
   useEffect(() => {
     listFetcher.load('/api/tags/search');
   }, []);
 
   const tags = listFetcher.data?.tags || [];
 
-  function reset() {
+  function onTagCreated(newTag: unknown) {
+    createFetcher.submit(
+      { id: newTag.id, name: newTag.label },
+      {
+        action: '/api/tags/add',
+        method: 'post',
+      }
+    );
     setSearch('');
-    setNewTagId(id());
+    setSelectedTags({ ...selectedTags, [newTag.id]: newTag });
+    listFetcher.load('/api/tags/search');
   }
 
+  const onTagSelected = (tag: unknown) => {
```

Comment:
As mentioned earlier, if we define a type `Tag`, then we could replace `(tag: unknown)` with `(tag: Tag)`.

Reply:
Thank you Tomas! I am working on it.